### PR TITLE
refactored the model monitoring scripts; used polars instead of pandas;

### PR DESCRIPTION
**Pull request by ranjan-stha**

### CODE REVIEW

File: core/models.py [hunk 1]
```diff
@@ -156,14 +156,17 @@ class Entry(BaseModel):
     original_entry_id = models.PositiveIntegerField(unique=True)
     lead = models.ForeignKey(Lead, on_delete=models.CASCADE)
     original_lang = models.CharField(max_length=50)
-    excerpt_en = models.TextField()
+    excerpt = models.TextField()
```
Comment (ranjan-stha) [inline core/models.py:159]: are we only pulling english excerpts ??

Comment (bewakes) [reply core/models.py:159]: In fact, we are pulling form the `excerpt` column of `entry` table. Whatever is there gets pulled without language information. Do you think we should store language info/translate accordingly?

Comment (ranjan-stha) [reply core/models.py:159]: I think it is okay to pull the excerpts irrespective of the language. The model is multi-lingual and able to generate predictions and embeddings (atleast for lang: en, fr, es, ar, portuguese). For other lang, preds and embeds might not be that good.

Comment (bewakes) [reply core/models.py:159]: Sounds good. Doesn't require any changes for now.

File: core/tasks/get_data.py [hunk 2]
```diff
@@ -31,6 +38,13 @@
 VERY_PAST_DATE = datetime(1990, 10, 10)  # Before creation of deep platform
 
 
+def get_projects_later_than_date():
+    try:
+        return datetime.strptime("%Y-%m-%d", FETCH_DEEP_PROJECTS_AFTER)
+    except ValueError:
+        return datetime(2021, 1, 1)
```
Comment (k9845) [inline core/tasks/get_data.py:46]: Lets add note here, why we are hard-coding.

File: nlp_scripts/model_monitoring/featuredrift.py [hunk 3]
```diff
@@ -130,19 +122,19 @@ def compute_feature_drift(
...
-    reference_data_df = pd.read_csv(reference_data_path)
-    current_data_df = pd.read_csv(current_data_path)
+    reference_data_df = pl.read_csv(reference_data_path)
+    current_data_df = pl.read_csv(current_data_path)
 
     feature_drift = FeatureDrift(reference_data_df, current_data_df)
-    df = feature_drift.compute_feature_drift(ref_n_samples=10, cur_n_samples=10)
+    df = feature_drift.compute_feature_drift(ref_n_samples=5, cur_n_samples=5)
     print(df)
+    print(df.columns)
```
Comment (k9845) [inline nlp_scripts/model_monitoring/featuredrift.py:140]:
```suggestion
```

File: nlp_scripts/model_monitoring/model_performance.py [hunk 4]
```diff
@@ -1,361 +1,425 @@
...
-    # print(df2)
-    df3 = modelperf.all_projects_perf_metrics()
-    # print(df3)
-    df4 = modelperf.calculate_ratios()
-    # print(df4)
-    df5 = modelperf.per_project_calc_ratios()
-    # print(df5)
+    modelperf = ModelPerformance(dataframe=pl_df)
+    modelperf.label_transform()
+    print(modelperf.projectwise_perf_metrics())
```
Comment (k9845) [inline nlp_scripts/model_monitoring/model_performance.py:425]:
```suggestion
```

File: nlp_scripts/model_monitoring/featuredrift.py (hunk 3 above)
Comment (bewakes) [reply nlp_scripts/model_monitoring/featuredrift.py:140]: It's inside __main__ so should not matter.

File: nlp_scripts/model_monitoring/model_performance.py (hunk 4 above)
Comment (bewakes) [reply nlp_scripts/model_monitoring/model_performance.py:425]: It's inside __main__ so should not matter.

File: core/tasks/get_data.py (hunk 2 above)
Comment (bewakes) [reply core/tasks/get_data.py:46]: Added.

**Review by k9845 — APPROVED** [review]
LGTM!

**Review by Rup-Narayan-Rajbanshi — APPROVED** [review]
LGTM

**Review by thenav56 — APPROVED** [review]

**Pull request merged**